templates/ModelDecoderEncoderSnippet.stencil

```diff
@@ -0,0 +1,31 @@
+{% if model.hasDateProperty %}
+
+enum CodingKeys: String, CodingKey {
+{% for property in model.properties %}
+       case {{ property.name }}
+{% endfor %}
+}
+   
+public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    {% for property in model.properties %}
+        {% if property.isDate %}
+        {% include "ModelDecoderDateProperty.stencil" %}
+        {% else %}
+        {% include "ModelDecoderProperty.stencil" %}
+        {% endif %}
+    {% endfor %}
+}
+
+public func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    {% for property in model.properties %}
+    {% if property.isDate %}
+    {% include "ModelEncoderDateProperty.stencil" %}
+    {% else %}
+    {% include "ModelEncoderProperty.stencil" %}
+    {% endif %}
+    {% endfor %}
+}
```

Comment:
It would probably be okay to split this out into a single ModelDecoderProperty file that works like a big switch. I doubt any individual type would need more than 10 LOC.